File: ShardsManager.API/ShardsManager.API/Models/CollectionStats.cs
using System;
using System.Collections.Generic;

namespace ShardsManager.API.Models
{
  public class CollectionStats
  {
    public long NoOfDocuments { get; set; }
    public double DataStorageSizeInMB { get; set; }
    public double IndexStorageSizeInMB { get; set; }
    public double TotalStorageSizeInMB { get; set; }
    public long NoOfIndexes { get; set; }
    public long NoOfChunks { get; set; }
    public List<ShardStats> ShardStats { get; set; }
  }

  public class ShardStats
  {
    public string ShardName { get; set; }
    public double StorageSizeInMB { get; set; }
    public double FreeStorageSizeInMB { get; set; }
    public long NoOfDocuments { get; set; }
    public long NoOfChunks { get; set; }
  }
}

File: ShardsManager.UI/src/app/shard-manager/models/collectionStats.ts
export interface CollectionStats {
  noOfDocuments: number;
  dataStorageSizeInMB: number;
  indexStorageSizeInMB: number;
  totalStorageSizeInMB: number;
  noOfIndexes: number;
  noOfChunks: number;
  shardStats: ShardStats[];
}

export interface ShardStats {
  shardName: string;
  storageSizeInMB: number;
  freeStorageSizeInMB: number;
  noOfDocuments: number;
  noOfChunks: number;
}
